### ISSUE TITLE
Getting Query Parameters in Codable Router

### ISSUE THREAD

**Issue by tettoffensive** [IBM-Swift/Kitura#1172]
```swift
router.get("/items") { (respondWith: ([Item]?, RequestError?) -> Void) in
    
}
```

How can I get any query parameters that came along with the request? In the last version before 2.0, I had access to the request object.

Also, It seems pretty limiting that I have to make my response be an array of Codable objects. I would like to wrap that inside another codable object. So, I'd love it to be `Item?` instead of `[Item]?` or support both ways.

**Comment by tunniclm** [comment]
I was just talking the other day about adding a single-object version of the Codable `get`, so look for a PR for this soon.

We are discussing the design of query parameters and path parameters with Codable routing right now and would welcome thoughts and feedback (and use cases) to help guide the design. Feel free to comment here or join us on our Slack (https://swift-at-ibm-slack.mybluemix.net/).

**Comment by seabaylea** [comment]
For the query parameters we're looking at a "type safe" implementation where you specify a struct/class of the allowable parameters (including optionals) which is pre-parsed and passed into your handler as a parameter.

We'll publish a "Kitura Evolution" proposal soon, but it would be good to get your feedback ahead of then if your happy to.

**Comment by tettoffensive** [comment]
I like the idea of type safe. Can't wait to see the proposal

**Comment by tunniclm** [comment]
The PR for single object get was merged (https://github.com/IBM-Swift/Kitura/pull/1176) and went into Kitura `2.1.0`

**Comment by ianpartridge** [comment]
Type-safe query parameters were released in Kitura 2.2
